packages/@ama-sdk/core/src/fwk/mocks/base-mock-adapter.ts

```diff
@@ -14,7 +14,7 @@ export abstract class BaseMockAdapter implements MockAdapter {
   protected mockFactory: (() => Promise<MockMap>) | undefined;
 
   constructor(
-    protected pathObjects: PathObject[],
+    protected pathObjects: PathObject[] | (() => Promise<PathObject[]>),
```

Comment:
should be `() => (PathObject[] | Promise<PathObject[]>)` to allow sync function as well

Reply:
Changed